### PR TITLE
Add GenerateEverything.hs, remove hand-written everything modules

### PR DESCRIPTION
**Pull request by ice1000**
#20

### CODE REVIEW

**Comment by ice1000** [comment]
I added `AllNonAsciiChars` by the way, because it looks good.

File: GenerateEverything.hs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import System.IO
+import System.Exit
+import System.FilePath
+import System.FilePath.Find
+
+headerFile     = "Header"
+allOutputFile  = "Everything"
+srcDirs        = ["Cubical/HITs", "Cubical/Basics"]
+
+excludedModules :: [FilePath]
```
Comment (mortberg) [inline GenerateEverything.hs]: I guess the modules listed here won't be listed in the Everything file?

Note that HInt has moved to Experiments now, so I don't think we have to list it. I would also not list HitInt and SmashProduct.

File: GenerateEverything.hs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  , "headers from library modules. The output is written to " ++ allOutputFile
+  , "with the file " ++ headerFile ++ " inserted verbatim at the beginning."
+  ]
+
+-- | Returns 'True' for all Agda files except for core modules.
+
+isLibraryModule :: FilePath -> Bool
+isLibraryModule f =
+  takeExtension f `elem` [".agda", ".lagda"]
+  -- && dropExtension (takeFileName f) /= "Core"
```
Comment (mortberg) [inline GenerateEverything.hs]: Why is this commented here? The comment above seems to indicate that it shouldn't be...

File: GenerateEverything.hs (hunk 1 above)
Comment (ice1000) [reply GenerateEverything.hs]: > Why SmashProduct

I see now. Updated.

File: GenerateEverything.hs (hunk 2 above)
Comment (mortberg) [reply GenerateEverything.hs]: So the "Returns ..." comment above is not correct anymore?

**Comment by mortberg** [comment]
I tried to build this locally, but it didn't work because I missed the `FileManip` package. The problem was then that I couldn't get the `FileManip` package to `cabal install` for me... I would prefer if the `agda/cubical` library did not rely on any external Haskell packages. It is already hard enough for people to get the development version of Agda up and running, so requiring them to install even more Haskell packages just to compile the library seems like a bad idea. 

Can you rewrite `GenerateEverything.hs` to be completely self-contained and not rely on any additional Haskell packages?  If not I prefer to have hand-written `Everything.hs` modules and just a simple bash script that checks that everything is included in the `Basics` and `HITs` directories.

File: GenerateEverything.hs (hunk 2 above)
Comment (ice1000) [reply GenerateEverything.hs]: Yes, I'll remove that.

**Comment by gallais** [comment]
Note that in the stdlib we have a [.cabal file](https://github.com/agda/agda-stdlib/blob/master/lib.cabal) which makes installing the required dependencies
to generate the `Everything` module fairly easy.

**Comment by ice1000** [comment]
@mortberg What if I add the .cabal file to this PR?

**Comment by mortberg** [comment]
> @mortberg What if I add the .cabal file to this PR? Is that gonna make this PR acceptable?

Please do it and I'll try it.  My fear is that it will still be too hard to compile things... I've seen how hard it is to just compile the development version of Agda on a variety of platforms and I don't want this library to be yet another obstacle for newcomers.

I think the ideal solution would be to have a file that checks if all the files has been added to the Everything.agda files. This can then be made part of the CI so that the developers can deal with potential issues without adding any extra burden for new users.

**Comment by gallais** [comment]
Potential solution: commit the `Everything.agda` generated by the haskell program?
Then only devs need to bother installing it.

**Comment by mortberg** [comment]
> Potential solution: commit the `Everything.agda` generated by the haskell program?
> Then only devs need to bother installing it.

How does the generated `Everything.agda` files deal with name clashes between modules?

**Comment by mortberg** [comment]
I closed this now as there has been no progress for many months. Feel free to open again if someone fixes it

**Pull request closed without merging**